eng/pipelines/runtimelab/runtimelab-post-build-steps.yml

```diff
@@ -51,18 +61,30 @@ steps:
             call $(Build.SourcesDirectory)\wasm-tools\emsdk\emsdk_env
             $(Build.SourcesDirectory)/src/tests/run$(scriptExt) runnativeaottests $(buildConfigUpper) ${{ parameters.archType }} ${{ parameters.osGroup }}
           displayName: Run WebAssembly tests in single file mode
+      - ${{ elseif in(parameters.platform, 'wasi_wasm_linux_x64_naot_llvm') }}:
+        - script: |
+            $(Build.SourcesDirectory)/src/tests/run$(scriptExt) --runnativeaottests ${{ parameters.archType }} $(buildConfigUpper) wasi
+          displayName: Run WebAssembly tests in single file mode
       - ${{ elseif eq(parameters.osGroup, 'windows') }}:
-        - script: $(Build.SourcesDirectory)/src/tests/run$(scriptExt) runnativeaottests $(buildConfigUpper) ${{ parameters.archType }}
+        - script: $(Build.SourcesDirectory)/src/tests/run$(scriptExt) runnativeaottests $(buildConfigUpper) ${{ parameters.archType }} ${{ parameters.osGroup }}
           displayName: Run tests in single file mode
       - ${{ else }}:
-        - script: $(Build.SourcesDirectory)/src/tests/run$(scriptExt) --runnativeaottests $(buildConfigUpper) ${{ parameters.archType }}
+        - script: $(Build.SourcesDirectory)/src/tests/run$(scriptExt) --runnativeaottests $(buildConfigUpper) ${{ parameters.archType }} ${{ parameters.osGroup }}
           displayName: Run tests in single file mode
 
     - ${{ if eq(parameters.archType, 'wasm') }}:
-      - script: |
-            call $(Build.SourcesDirectory)\wasm-tools\emsdk\emsdk_env
-            $(Build.SourcesDirectory)/build$(scriptExt) libs.tests -test -a ${{ parameters.archType }} -os ${{ parameters.osGroup }} -lc ${{ parameters.librariesConfiguration }} -rc $(buildConfigUpper) /p:TestNativeAot=true /p:RunSmokeTestsOnly=true
-        displayName: Build and run WebAssembly libraries tests
+      - ${{ if in(parameters.platform, 'browser_wasm_win', 'wasi_wasm_win') }}:
+        - script: |
+              call $(Build.SourcesDirectory)\wasm-tools\emsdk\emsdk_env
+              $(Build.SourcesDirectory)/build$(scriptExt) libs.tests -test -a ${{ parameters.archType }} -os ${{ parameters.osGroup }} -lc ${{ parameters.librariesConfiguration }} -rc $(buildConfigUpper) /p:TestNativeAot=true /p:RunSmokeTestsOnly=true
+          displayName: Build and run WebAssembly libraries tests
+      - ${{ else }}:
+        - script: |
+              source $(Build.SourcesDirectory)/wasm-tools/emsdk/emsdk_env.sh
+              $(Build.SourcesDirectory)/build$(scriptExt) libs.tests -test -a ${{ parameters.archType }} -os ${{ parameters.osGroup }} -lc ${{ parameters.librariesConfiguration }} -rc $(buildConfigUpper) /p:TestNativeAot=true /p:RunSmokeTestsOnly=true
+          displayName: Build and run WebAssembly libraries tests
```

Comment:
Why does this need to be separate? `emsdk_env.sh` is already part of `build.sh`.